examples/e-commerce/src/app/snippet/snippet.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Component, Input } from '@angular/core';
+
+@Component({
+  selector: 'app-snippet',
+  template: `<span class="ais-Snippet" [innerHtml]="content"></span>`,
+})
+export class Snippet {
+  @Input() attribute: string;
+  @Input() hit: { _snippetResult?: {}; label?: string; highlighted?: string };
+  @Input() tagName: string = 'em';
+
+  get content() {
+    if (this.hit.hasOwnProperty('_snippetResult')) {
+      const attributeHighlighted = this.hit._snippetResult[this.attribute];
+
+      // check that the attributeHighlighted is a string
+      if (
+        attributeHighlighted !== undefined &&
+        typeof attributeHighlighted.value === 'string'
+      ) {
+        return this.replaceWithTagName(attributeHighlighted.value);
+      }
+    }
+
+    const fallback = this.hit[this.attribute];
+    if (!fallback) {
+      console.warn(
+        `Could not find attribute [${
+          this.attribute
+        }] into hit object, will display an empty string.`
+      );
+
+      return '';
+    }
+
+    return fallback;
+  }
+
+  replaceWithTagName(value: string) {
```

Comment:
I don't think this works as expected because InstantSearch.js 3 returns `mark` instead of `em`. Ideally we should use the [InstantSearch.js `snippet` helper](https://www.algolia.com/doc/api-reference/widgets/snippet/js/) (the Angular `ais-highlight` widget seems broken on the v3 branch for this reason).

Reply:
Right, I've changed it to use the snippet function from instantsearch.js.
I will try to fix the problem in highlight widget in a separate PR.
fe05cbd